File: JavaScript/8js 3/noBoringZeros.js
/*
|--------------------------------------------------------------------------
| No zeros for heros
|--------------------------------------------------------------------------
|
*/

function noBoringZeros(n) {
   while (n % 10 == 0 && n != 0) {
      n /= 10;
   }
   return n;
}


function noBoringZeros(n) {
   return Number(String(n).replace(/0+$/, ""));
}


function noBoringZeros(n) {
   return n % 10 || n === 0 ? n : noBoringZeros(n / 10);
}


console.log(noBoringZeros(1450), 145);
console.log(noBoringZeros(960000), 96);
console.log(noBoringZeros(1050), 105);
console.log(noBoringZeros(-1050), -105);
console.log(noBoringZeros(-105), -105);
console.log(noBoringZeros(0), 0);

File: Python/8py 2/no_boring_zeros.py
"""
|--------------------------------------------------------------------------
| No zeros for heros
|--------------------------------------------------------------------------
|
"""

def no_boring_zeros(n):
   while n % 10 == 0 and n != 0:
      n /= 10
   return n


def no_boring_zeros(n):
   try:
      return int(str(n).rstrip('0'))
   except ValueError:
      return 0


def no_boring_zeros(n):
   return int(str(n).strip('0') or 0)


print(no_boring_zeros(1450), 145)
print(no_boring_zeros(960000), 96)
print(no_boring_zeros(1050), 105)
print(no_boring_zeros(-1050), -105)
print(no_boring_zeros(0), 0)
print(no_boring_zeros(2016), 2016)